File: +ndi/+cloud/+sync/+internal/updateFileInfoForLocalFiles.m
function document = updateFileInfoForLocalFiles(document, fileDirectory)
% updateFileInfoForLocalFiles - Update file info of document for local files
%
% Syntax:
%   document = ndi.cloud.sync.internal.updateFileInfoForLocalFiles(document, fileDirectory)
%       updates the file info of the document to point to a file in the
%       provided (local) file directory
%
% Input Arguments: 
%   document - The document object that contains file info to be updated
%   fileDirectory - The directory where local files are stored
%
% Output Arguments: 
%   document - The updated document object with new file info
%
% See also:
%   ndi.cloud.sync.internal.updateFileInfoForRemoteFiles

    if document.has_files()
        originalFileInfo = document.document_properties.files.file_info;
        document = document.reset_file_info();
    
        for i = 1:numel(originalFileInfo)
            file_uid = originalFileInfo(i).locations(1).uid;
            file_location = fullfile(fileDirectory, file_uid);
    
            filename = originalFileInfo(i).name; % name for ingestion
            if isfile(file_location)
                document = document.add_file(filename, file_location);
            else
                warning('Local file does not exist for document "%s"', ...
                    document.document_properties.base.id)
            end
        end
    end

    

end

File: ndi-python/ndi/cloud/sync/internal/update_file_info_for_local_files.py
"""Update file info of document for local files.

Ported from: ndi.cloud.sync.internal.updateFileInfoForLocalFiles (MATLAB)
"""

import os
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ndi.document import Document


def update_file_info_for_local_files(document: 'Document', file_directory: str) -> 'Document':
    """Update file info of document for local files.

    Updates the file info of the document to point to a file in the provided
    (local) file directory.

    Args:
        document: The document object that contains file info to be updated
        file_directory: The directory where local files are stored

    Returns:
        The updated document object with new file info
    """
    if document.has_files():
        original_file_info = document.document_properties.get('files', {}).get('file_info', [])
        document = document.reset_file_info()

        for info in original_file_info:
            locations = info.get('locations', [])
            if locations:
                file_uid = locations[0].get('uid')
                if file_uid:
                    file_location = os.path.join(file_directory, file_uid)

                    filename = info.get('name')  # name for ingestion
                    if os.path.isfile(file_location):
                        document = document.add_file(filename, file_location)
                    else:
                        print(f'Warning: Local file does not exist for document '
                              f'"{document.document_properties["base"]["id"]}"')

    return document
